### PR TITLE
refactor: use concepts even more

### PR DESCRIPTION
**Pull request by scarf005**
### Required

- [x] I wrote the PR title in [conventional commit format](https://docs.cataclysmbn.org/en/contribute/changelog_guidelines/).
- [x] I ran the [code formatter](https://docs.cataclysmbn.org/en/contribute/contributing/#code-style).
- [x] I linked any relevant issues using [github keyword syntax](https://docs.cataclysmbn.org/en/contribute/contributing/#pull-request-notes) so it can be closed automatically.
- [x] I have [committed my changes to new branch that isn't `main`](https://docs.cataclysmbn.org/en/contribute/contributing/#make-your-changes) so it won't cause conflict when updating `main` branch later.

## Purpose of change

followup to #5537: apply https://clang.llvm.org/extra/clang-tidy/checks/modernize/use-constraints.html

## Describe the solution

ran:

```
$ parallel ./build-scripts/clang-tidy-wrapper.sh {} -fix '--checks=-\*,modernize-use-constraints,modernize-type-traits' ::: src/*.{cpp,h}
```

btw it took 370s lol

## Testing

should (probably) build

### CODE REVIEW

**Comment by OrenAudeles** [comment]
a few build errors on this one still :(

File: src/fmtlib_format.h [hunk 1]
```diff
@@ -4359,7 +4359,7 @@ make_format_to_n_args( const Args &... args )
     return format_arg_store<buffer_context<Char>, Args...>( args... );
 }
 
-template < typename Char, enable_if_t < ( !std::is_same<Char, char>::value ), int >>
+template < typename Char, enable_if_t < ( !std::is_same_v<Char, char> ), int >>
```
Comment (OrenAudeles) [inline src/fmtlib_format.h:4362]: Compilers are angry, so I think this change from `!is_same<>::value` to `!is_same_v<>` needs to also be reflected here https://github.com/cataclysmbnteam/Cataclysm-BN/blob/a3c18afd6e62ed7d55be1d4a3c13aab5936b80b7/src/fmtlib_core.h#L2188-L2191

Comment (scarf005) [reply src/fmtlib_format.h:4361]: applied in [`b535e4c` (#5538)](https://github.com/cataclysmbnteam/Cataclysm-BN/pull/5538/commits/b535e4c69a59ea0b99439e5bd21e85d196f05fff)

**Review by chaosvolt — APPROVED** [review]
It's a lot to go over but it compiles and loads, and only tests that're failing are clang whining about unrelated things again: https://github.com/cataclysmbnteam/Cataclysm-BN/actions/runs/11241032936/job/31251730123?pr=5538

**Pull request merged**